Create doxy-graph-mode

gustavopuche opened the pull request:
### Brief summary of what the package does

doxy-graph-mode links source code to doxygen call graphs.

It allows to interactively see the call graph or the inverted call graph of a given function or method from source code.

### Direct link to the package repository

https://github.com/gustavopuche/doxy-graph-mode

### Your association with the package

I am the author and the maintainer of doxy-graph-mode.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [ x ] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [ x ] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ x ] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [ x ] My elisp byte-compiles cleanly
- [ x ] `M-x checkdoc` is happy with my docstrings
- [ x ] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I have confirmed some of these without doing them

riscy commented:
Quick first pass -- thanks for your close adherence to the checklist.

### doxy-graph-mode.el ###

byte-compile (using Emacs 26.3):
- Per CONTRIBUTING.org, consider using lexical binding.
  See: https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org#fixing-typical-problems
- Lexical binding will be used in what follows
- No issues!

checkdoc (using version 0.6.1):
- No issues!

package-lint (using version 20200419.406):
- No issues!

### Package ###

<!-- LICENSE.md excerpt: `### GNU GENERAL PUBLIC LICENSE...` -->
- GitHub API found `GNU General Public License v3.0`
- Add license boilerplate or an [SPDX-License-Identifier](https://spdx.org/using-spdx-license-identifier) to doxy-graph-mode.el

Suggestions/experimental static checks:
- doxy-graph-mode.el#L151: It's safer to sharp-quote function names; use `#'`
- doxy-graph-mode.el#L154: It's safer to sharp-quote function names; use `#'`

Finally on line 151 for example you are autoloading an `add-hook` onto C and C++ mode -- instead of this, you should take a look at `define-globalized-minor-mode` (and check out its TURN-ON argument)

gustavopuche commented:
Hi all,

I have done all this modifications.

I have removed the add-hook features and I have updated the README.org 
file with setup instructions.

Bests regards,

Gustavo Puche

El 02/08/2020 a las 1:28, Chris Rayner escribió:
>
> Quick first pass -- thanks for your close adherence to the checklist.
>
>
>       doxy-graph-mode.el
>
> byte-compile (using Emacs 26.3):
>
>   * Per CONTRIBUTING.org, consider using lexical binding.
>     See:
>     https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org#fixing-typical-problems
>   * Lexical binding will be used in what follows
>   * No issues!
>
> checkdoc (using version 0.6.1):
>
>   * No issues!
>
> package-lint (using version 20200419.406):
>
>   * No issues!
>
>
>       Package
>
>   * GitHub API found |GNU General Public License v3.0|
>   * Add license boilerplate or an SPDX-License-Identifier
>     <https://spdx.org/using-spdx-license-identifier> to doxy-graph-mode.el
>
> Suggestions/experimental static checks:
>
>   * doxy-graph-mode.el#L151: It's safer to sharp-quote function names;
>     use |#'|
>   * doxy-graph-mode.el#L154: It's safer to sharp-quote function names;
>     use |#'|
>
> Finally on line 151 for example you are autoloading an |add-hook| onto 
> C and C++ mode -- instead of this, you should take a look at 
> |define-globalized-minor-mode| (and check out its TURN-ON argument)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/melpa/melpa/pull/7049#issuecomment-667599842>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ACEDGZ2FYYPTMNB2YDRNMQ3R6SQJZANCNFSM4PKNK34Q>.
>

riscy commented:
Looks good!  I spotted a typo at the top of the el file (Copiright -> Copyright) but nothing to worry about.
